FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -1134,6 +1133,41 @@ static BaseType_t prvDownloadLoop( void )
         }
         else
         {
+            if( xDownloadRespItem.xResponse.statusCode != httpexampleHTTP_STATUS_CODE_PARTIAL_CONTENT )
+            {
+                LogError( ( "Received response with unexpected status code: %d.", xDownloadRespItem.xResponse.statusCode ) );
+                xIsConnectionEstablished = pdFALSE;
+                xStatus = pdFAIL;
+                break;
+            }
+
+            /* Check for the "Connection: close" here (instead of the response
+             * task), so that the failed request can be re-sent */
+            if( xDownloadRespItem.xResponse.respFlags == HTTP_RESPONSE_CONNECTION_CLOSE_FLAG )
+            {
+                LogInfo( ( "'Connection:close' header found in server response. Attempting to re-connect to the server." ) );
+                vTaskSuspend( xRequestTask );
+                vTaskSuspend( xResponseTask );
+
+                /* Disconnect and re-establish the connection. */
+                TLS_FreeRTOS_Disconnect( &xNetworkContext );
+                xStatus = connectToServerWithBackoffRetries( prvConnectToServer,
+                                                             &xNetworkContext );
+
+                vTaskResume( xRequestTask );
+                vTaskResume( xResponseTask );
+
+                if( xStatus != pdPASS )
+                {
+                    /* Notify the response task that a response should not be expected. */
+                    xTaskNotify( xResponseTask, httpexampleHTTP_FAILURE, eSetBits );
+                    xIsConnectionEstablished = pdFALSE;
+
+                    LogError( ( "Could not reconnect to server. Exiting task." ) );
+                    break;
+                }
+            }
+
             LogInfo( ( "The HTTP task received a response from the server. Adding to response queue." ) );
 
             /* Add response to response queue. */
```

Comment:
I think it would be useful for the response task (prvResponseTask) to still get a non-206 response to print out the error message in the response body sent by S3.